### PR TITLE
PARQUET-1964: FOLLOWUP: Avoid constructing useless ArrayList

### PR DESCRIPTION
**Pull request by wangyum**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java [hunk 1]
```diff
@@ -827,7 +827,7 @@ public long getRecordCount() {
   }
 
   public long getFilteredRecordCount() {
-    if (!options.useColumnIndexFilter()) {
+    if (!options.useColumnIndexFilter() && options.getRecordFilter() != null) {
```
Comment (wangyum) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java:830]: How to reproduce this issue:
```scala
val hadoopInputFile = HadoopInputFile.fromPath(new Path("/path/to/parquet/000.snappy.parquet"), new Configuration())
val reader = ParquetFileReader.open(hadoopInputFile)
val recordCount = reader.getFilteredRecordCount
reader.close()
```

**Comment by wangyum** [comment]
cc @gszadovszky

**Comment by gszadovszky** [comment]
Thanks a lot for working on this. It is a good catch!

I've had to investigate a bit why we do not catch this in the unit tests. The answer is that in most cases a [NOOP](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/filter2/compat/FilterCompat.java#L62) filter is created instead of having a `null` value. We use this `NOOP` if no filter is specified in the builder/config. Meanwhile it is possible to set a `null` in the builder/config so the NPE may occur.
I've also realized that in older filter implementations (row group level min/max, dictionary) the `NOOP` filter is fine and do not have significant performance costs over a `null` check. Meanwhile for column index of bloom filter based filtering do have performance costs because they read the related data from the file even for `NOOP`.

If you don't mind, @wangyum, I would like to take care of this (including the NPE and the potential performance issues).

**Comment by wangyum** [comment]
@gszadovszky Please take care of this.

**Comment by wangyum** [comment]
cc @gszadovszky

**Review by gszadovszky — APPROVED** [review]
Thanks for fixing this. LGTM.

There are a couple unrelated Travis failures for our PRs. Let's wait for a green build.

**Review by Fokko — APPROVED** [review]

**Pull request merged**